fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingStore.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding
+
+import mozilla.components.lib.state.Action
+import mozilla.components.lib.state.State
+import mozilla.components.lib.state.Store
+
+/**
+ * The [Store] for holding the [OnboardingFragmentState] and applying [OnboardingAction]s.
+ */
+class OnboardingStore(
+    initialState: OnboardingFragmentState = OnboardingFragmentState(),
+) : Store<OnboardingFragmentState, OnboardingAction>(
+    initialState = initialState,
+    reducer = ::onboardingFragmentStateReducer,
+)
+
+/**
+ * The state used for managing [OnboardingFragment].
+ *
+ * @property onboardingState Describes the onboarding account state.
+ */
+data class OnboardingFragmentState(
+    val onboardingState: OnboardingState = OnboardingState.SignedOutNoAutoSignIn,
+) : State
```

Comment:
This is obviously where the naming can get complicated, as we have `State` in our Redux pattern but it's also being used as a data class wrapper for a state machine in Onboarding.

Reply:
Ya, the naming gets a bit tricky.